soc/xtensa/esp32s2/soc_cache.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2021 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "soc.h"
+
+void IRAM_ATTR esp_config_instruction_cache_mode(void)
+{
+	cache_size_t cache_size;
+	cache_ways_t cache_ways;
+	cache_line_size_t cache_line_size;
+
+#if CONFIG_ESP32S2_INSTRUCTION_CACHE_8KB
+	esp_rom_Cache_Allocate_SRAM(CACHE_MEMORY_ICACHE_LOW, CACHE_MEMORY_INVALID,
+				CACHE_MEMORY_INVALID, CACHE_MEMORY_INVALID);
+	cache_size = CACHE_SIZE_8KB;
+#else
+	esp_rom_Cache_Allocate_SRAM(CACHE_MEMORY_ICACHE_LOW, CACHE_MEMORY_ICACHE_HIGH,
+				CACHE_MEMORY_INVALID, CACHE_MEMORY_INVALID);
+	cache_size = CACHE_SIZE_16KB;
+#endif
+
+	cache_ways = CACHE_4WAYS_ASSOC;
+
+#if CONFIG_ESP32S2_INSTRUCTION_CACHE_LINE_16B
+	cache_line_size = CACHE_LINE_SIZE_16B;
+#else
+	cache_line_size = CACHE_LINE_SIZE_32B;
+#endif
```

Comment:
Use a define to refactor this so it looks like

cache_line_size = ESP32S2_CACHE_LINE_SIZE;

The same applies many other places in this file.

Reply:
@SebastianBoe, thanks for the suggestions.

I have refactored most of it, except for the macro conditionals enclosing `esp_rom_Cache_Allocate_SRAM` since the refactoring of SRAM's layout definitions felt clunkier than in the original code.

Please, take a look.